### ISSUE TITLE
[MapView] 'view' prop fails with `Converting circular structure to JSON` error

### ISSUE THREAD

**Issue by af7** [facebook/react-native#5098]
I have the latest v0.17.0 version that supports the `view` prop.

Unfortunately, I get the following error:

`Converting circular structure to JSON`

This is the code that I am using (which is actually from your own example page):

````
        let myView = (<View style={{
          alignItems: 'center',
        }}>
            <Text style={{fontWeight: 'bold', color: '#f007'}}>
                Thumbs Up!
            </Text>
            <Image
                style={{width: 90, height: 65, resizeMode: 'cover'}}
                source={require('../assets/marker.png')}
            />
        </View>);

        let annotation = [{
            latitude: 34,
            longitude: 22,
            title: 'Thumbs Up!',
            //image: require('../assets/marker.png'),
            view: myView,
        }];

        return (
            <View>
                <MapView
                    style={styles.map}
                    region={region}
                    annotations={annotation}
                />
           </View>
````

Note that if i remove the `view` prop and use an `image` prop instead, everything is OK.

Is this a bug?

**Comment by af7** [comment]
Actually it is a bug. 

I see that you made some fixes after the release of 1.17.0 to make it work (but it has other problems in my case). All docs describe the new 'view' functionality as available but this is not the case; the current release at NPM does not support it (without us really knowing it).

I looked at the history of commits and made the "map-related" changes myself (based on those commits), knowing that this is NOT the proper process. This fixed the `Converting circular structure to JSON` error, however I get a new error now within the XCODE environment (execution pauses at `RCTComponentData.m` on the `propBlockForKey` block).

I also tried to use the latest `master` branch but this did not work for other reasons.

Can you please make a patch release so that the MapView works as described on your docs?

Thank you.

**Comment by jaygarcia** [comment]
I just ran into this myself. Does MapView only work with local images for pins?
/cc @browniefed (i saw that you may have worked on this)

**Comment by browniefed** [comment]
@jaygarcia not a ton of experience in the realm of the popups, I would just recommend upgrading to .18 as you can now use any view for an annotation. I wouldn't recommend using master, but just using the .18-rc branch @af7

**Comment by jaygarcia** [comment]
Thanks @browniefed =).  Just an FYI, this is happening in the .18-rc release as well.

Any ideas on who added those features to MapView?

**Comment by browniefed** [comment]
@jaygarcia nick wood, you can ping him, I'm sure he is constantly pinged haha

**Comment by af7** [comment]
Hi, i am the OP, it seems that it still does not work. Did you find anything new on this? Thanks!

Also, this is related to the latest changes, I hope this can be fixed too:
https://github.com/facebook/react-native/issues/5111

**Comment by jaygarcia** [comment]
@af7 , it seems to work well for me -- i'm using 18.0 - rc

![2016-01-11_12-15-31](https://cloud.githubusercontent.com/assets/180031/12240731/1af13738-b85d-11e5-9523-201e066e1689.png)

**Comment by af7** [comment]
@jaygarcia Are these markers *views*? Or just images? Because the problem is with using a *view* for a marker, please let me know. Thanks.

**Comment by jedlau** [comment]
I was seeing this error, too, on 0.17.0. I can confirm that it's fixed in 0.18.0-rc. Yes, I'm using `view`, not `image`, for the annotation.

```javascript
      annotations.push({
        latitude: lat,
        longitude: lng,
        title: name,
        subtitle: detailedName,
        view:
          <View style={{ alignItems: "center" }}>
            <Image
              source={require("map-marker.png")}
              style={{height: 25, width: 25 }}
            />
          </View>
      });
```

**Comment by af7** [comment]
I can also confirm that this is now fixed (0.18), so we can close the issue if you wish.

**Comment by browniefed** [comment]
@af7 you can close it your self since you created it
